spec/tbd_tests_spec.rb

```diff
@@ -11543,4 +11549,570 @@
+
+    io[:edges].each do |edge|
+      expect(edge.key?(:surfaces        )).to be( true)
+      expect(edge[:surfaces].is_a?(Array)).to be( true)
+      expect(edge[:surfaces].empty?      ).to be(false)
+
+      next unless edge[:surfaces].include?(leftdoor)
+
+      expect(edge.key?(:type  )).to be(true)
+      expect(edge.key?(:length)).to be(true)
+
+      door_grade += 1 if edge[:type].to_s.include?("grade"     )
+      door_heads += 1 if edge[:type].to_s.include?("head"      )
+      door_sills += 1 if edge[:type].to_s.include?("sill"      )
+      door_jambs += 1 if edge[:type].to_s.include?("jamb"      )
+      door_trns  += 1 if edge[:type].to_s.include?("transition") # shared
+
+      door_head_m += edge[:length] if edge[:type].to_s.include?("head"      )
+      door_sill_m += edge[:length] if edge[:type].to_s.include?("sill"      )
+      door_jamb_m += edge[:length] if edge[:type].to_s.include?("jamb"      )
+      door_trns_m += edge[:length] if edge[:type].to_s.include?("transition")
+    end
+
+    # Again, 5x edges (instead of original 4x).
+    expect(door_grade).to eq(1)
+    expect(door_heads).to eq(0) # now shared with transom (see transition)
+    expect(door_sills).to eq(0) # overriden as grade
+    expect(door_jambs).to eq(2) # 1x full height + 1x partial height
+    expect(door_trns ).to eq(2) # shared with sidelight + transom
+
+    expect(door_jamb_m).to be_within(0.01).of(maxZ + 0.9)
+    expect(door_trns_m).to be_within(0.01).of(maxZ - 0.9 + maxY - minY)
+
+    io[:edges].each do |edge|
+      expect(edge.key?(:surfaces)        ).to be( true)
+      expect(edge[:surfaces].is_a?(Array)).to be( true)
+      expect(edge[:surfaces].empty?      ).to be(false)
+
+      next unless edge[:surfaces].include?(trnsom)
+
+      expect(edge.key?(:type  )).to be(true)
+      expect(edge.key?(:length)).to be(true)
+
+      trsm_heads += 1 if edge[:type].to_s.include?("head"      )
+      trsm_sills += 1 if edge[:type].to_s.include?("sill"      )
+      trsm_jambs += 1 if edge[:type].to_s.include?("jamb"      )
+      trsm_trns  += 1 if edge[:type].to_s.include?("transition") # shared
+
+      trsm_head_m += edge[:length] if edge[:type].to_s.include?("head"      )
+      trsm_sill_m += edge[:length] if edge[:type].to_s.include?("sill"      )
+      trsm_jamb_m += edge[:length] if edge[:type].to_s.include?("jamb"      )
+      trsm_trns_m += edge[:length] if edge[:type].to_s.include?("transition")
+    end
+
+    # 5x edges (instead of original 4x).
+    expect(trsm_heads).to eq(1)
+    expect(trsm_sills).to eq(0) # instead shared with door and sidelight
+    expect(trsm_jambs).to eq(2) # 1x full height + 1x partial height
+    expect(trsm_trns ).to eq(2) # shared with sidelight + door
+
+    expect(trsm_jamb_m).to be_within(0.01).of(2 * 0.4          )
+    expect(trsm_trns_m).to be_within(0.01).of(maxY - minY + 0.5)
+    expect(trsm_head_m).to be_within(0.01).of(trsm_trns_m      )
+  end
+
+  it "can check for conditioned attics" do
```

Comment:
Tests illustrating how one may (temporarily) switch from an unconditioned attic to an indirectly conditioned plenum (or any other free-floating or dead space).